File: API/Entities/Product.cs
namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public string Type { get; set; }       
        public string Color { get; set; }
        public long Price { get; set; }
        public decimal? Weight { get; set; }        
        public int Stock { get; set; } 
        public string ImageUrl { get; set; }
        public string Description { get; set; }
        
    }
}

File: client/src/app/model/product.ts
export interface Product {
  id: number;
  brand: string;
  model: string;
  type: string;
  color: string;
  price: number;
  weight?: number;
  stock: number;
  imageUrl: string;
  description: string;
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  types: string[];
  brands: string[];
  pageNumber: number;
  pageSize: number;
}
